test/fstapp_SUITE.erl

```diff
@@ -4,26 +4,49 @@
 
 -compile(export_all).
 
-init_per_suite(Config) ->
-	application:ensure_all_started(fstapp),
+init_per_testcase(_, Config) ->
+	true = register(fstapp_SUITE_process, self()),
+	_ = application:ensure_all_started(fstapp),
 	Config.
 
-end_per_suite(Config) ->
-	application:stop(fstapp),
-	Config.
+end_per_testcase(_, _Config) ->
+	application:stop(fstapp).
 
 all() ->
-	[my_test_case].
-
-my_test_case(_Config) ->
-
-	ct:log(start, "The server start by collecting metrics with frequency value as 5ms~n"),
-	CurrentFreq = 5000, % It represents the time in milliseconds.
-	CurrentFreq = fstapp:get_current_frequency(),
-
-	ct:log(change_frequency, "Update the frequency to 10ms~n"),
-	NewFreq = 10000,
-	ok = fstapp:change_frequency(NewFreq),
-	NewFreq = fstapp:get_current_frequency().
-
-
+        [metrics_test, frequency_test, stop_metrics_test].
+
+metrics_test(_Config) ->
+
+        receive
+                Metrics ->
+                	{cpu, _} = lists:keyfind(cpu, 1, Metrics),
+                        {ostype, _} = lists:keyfind(ostype, 1, Metrics),
+                        {proc, _} = lists:keyfind(proc, 1, Metrics),
+                        {disk, _} = lists:keyfind(disk, 1, Metrics)
+	after
+		10000 ->
+			ct:fail("Metrics are not received or handler did not send a message.")
+        end.
+
+
+frequency_test(_Config) ->
+
+        ct:log(start, "The server start by collecting metrics with frequency value as 5ms~n"),
+        CurrentFreq = 5000, % It represents the time in milliseconds.
+        CurrentFreq = fstapp:get_current_frequency(),
+
+        ct:log(change_frequency, "Update the frequency to 10ms~n"),
+        NewFreq = 10000,
+        ok = fstapp:change_frequency(NewFreq),
+        NewFreq = fstapp:get_current_frequency().
+
+stop_metrics_test(_Config) ->
+
+	ok = fstapp:stop_metrics(),
+	receive
+		Metrics ->
+			ct:fail("The server do not stop to collect the metrics.")
+	after
+		10000 ->
```

Comment:
I like you test this too but waiting for 10 seconds will make your tests slow. What if you update the frequency to 1 second and just wait for 2 or 3?